src/lib/components/Header/Nav/NavMenu.svelte

```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+  import Link from "$lib/components/Link/Link.svelte";
+  import chunk from "$lib/util/chunk";
+  import classNames from "$lib/util/classNames";
+  import { createEventDispatcher } from "svelte";
+  import type { NavMenuProps, NavLinkType } from "./types";
+  type $$Props = NavMenuProps;
+
+  const dispatch = createEventDispatcher();
+
+  export let id = "";
+  export let current = false;
+  export let columns = 1;
+  export let children: NavLinkType[] = [];
+  export let expanded = false;
+
+  $: buttonClassNames = classNames("usa-accordion__button usa-nav__link", current && "usa-current");
+
+  // If we want more than one column, evenly divide the nav items into mega menu columns.
+  // TODO: We may want to let the CMS determine the position of each item in the future.
+  let megaMenuColumns: NavLinkType[][] = [];
+  $: {
+    if (columns > 1) {
+      // Calculate the maximum length of columns (maximum number of items in a column) and provide
+      //   to chunk as the size (second) parameter.
+      megaMenuColumns = chunk(children, Math.ceil(children.length / columns));
+    }
+  }
```

Comment:
@Benaiah I opted to change this from `childColumns` to `megaMenuColumns` to denote that it's only being used as a mega menu. I kept most of your changes around setting the default `columns` to `1` but checked that that value was greater than `1` before setting `megaMenuColumns`. I also edited and added comments here to make it clearer what was happening.

Reply:
Sounds good, that's a good choice of variable name.